apps/user-ms/src/main/java/it/pagopa/selfcare/user/service/UserServiceImpl.java

```diff
@@ -33,4 +53,39 @@ public Multi<UserProductResponse> getUserProductsByInstitution(String institutio
                                 .build()))
                 .merge();
     }
+
+    @Override
+    public Uni<UserResource> retrievePerson(String userId, String productId, String institutionId) {
+        Map<String, Object> queryParameter = buildQueryParams(userId, productId, institutionId);
+        return userInstitutionService.retrieveFirstFilteredUserInstitution(queryParameter)
+                .onItem().ifNull().failWith(() -> {
+                    log.error(String.format(USER_NOT_FOUND_ERROR.getMessage(), userId));
+                    return new ResourceNotFoundException(String.format(USER_NOT_FOUND_ERROR.getMessage(), userId), USER_NOT_FOUND_ERROR.getCode());
+                })
+                .onItem().transformToUni(userInstitution -> userRegistryApi.findByIdUsingGET(USERS_WORKS_FIELD_LIST, userInstitution.getUserId()))
+                .onFailure(this::checkIfNotFoundException).transform(t -> new ResourceNotFoundException(String.format(USER_NOT_FOUND_ERROR.getMessage(), userId), USER_NOT_FOUND_ERROR.getCode()));
```

Comment:
You could name '**t**' as isNotFound....In addition, pay attention that here all exceptions will be throw as ResourceNotFoundException, you don't check if t is true or false.